src/scry/context.cr

```diff
@@ -27,26 +29,32 @@ module Scry
     # See, https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#shutdown-request
     # NOTE: For some reason the client doesn't accept `ResponseMessage.new(nil)` on shutdown even with emit_null.
     def dispatch(msg : RequestMessage)
-      exit(0) if msg.method == "shutdown"
       Log.logger.debug(msg.method)
-      if params = msg.params
-        dispatchRequest(params, msg)
-      end
+      dispatch_request(msg.params, msg)
     end
 
     def dispatch(msg : NotificationMessage)
       Log.logger.debug(msg.method)
-      dispatchNotification(msg.params, msg)
+      dispatch_notification(msg.params, msg)
+    end
+
+    private def dispatch_request(params : Nil, msg)
+      if msg.method == "shutdown"
+        Scry.shutdown = true
+        ResponseMessage.new(msg.id, nil)
+      elsif msg.method == "exit"
```

Comment:
I think the 'exit' handling needs to be in dispatch_notification as it is a notification message not a request message (unlike shutdown which is in the correct place)

Reply:
Oh, yeah, Thank you!